src/Ormolu/Printer/Meat/Declaration/Signature.hs

```diff
@@ -123,7 +123,7 @@ p_inlineSig name InlinePragma {..} = pragmaBraces $ do
     ConLike -> txt "CONLIKE"
     FunLike -> return ()
   space
-  p_activation inl_act
+  when (inl_act /= NeverActive) $ p_activation inl_act
```

Comment:
This is necessary as for some reason, this code
```haskell
{-# NOINLINE foo #-}
```
is parsed as having activation `NeverActive`, but 
```haskell
{-# NOINLINE [~] foo #-}
```
is invalid. This does not happen with e.g. `INLINE`, and is also not fixed by GHC 9.2. Maybe this is worth reporting/asking if it is intentional?

Reply:
Yeah, it is worth asking. I believe the issue tracker lives here now: https://gitlab.haskell.org/ghc/ghc/-/issues

Reply:
https://gitlab.haskell.org/ghc/ghc/-/issues/20603

Reply:
I got an explanation in the GHC issue, this is intentional behavior: the activation is not related to the `InlineSpec` (which is `NoInline` in this case), so `NeverActive` for this pragma does *not* mean that (as I thought and found confusing) that `NOINLINE` is never active. Rather, it means that inlining is never active.